src/dmd/expressionsem.d

```diff
@@ -3706,6 +3717,15 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
 
                 Expression arg = (*exp.arguments)[i];
                 arg = resolveProperties(sc, arg);
+                if (arg.op == TOK.type)
+                {
+                    arg = resolveAliasThis(sc, arg);
+                    if (arg.op == TOK.type)
+                    {
+                        exp.error("new can only create structs, dynamic arrays or class objects, not `%s`'s", originalNewtype.toChars());
```

Comment:
what about integers, floats, and unions?

I think the message should read "cannot `new` a `%s`" or something like it.

Reply:
Just committed a new version... does that look better to you?